app/assets/javascripts/discourse/app/lib/user-menu/base-item.js

```diff
@@ -30,6 +38,20 @@ export default class UserMenuBaseItem {
 
   get topicId() {}
 
+  get avatarTemplate() {}
+
+  get iconComponent() {
+    return this.siteSettings.show_user_menu_avatars ? UserMenuIconAvatar : null;
+  }
+
+  get iconComponentArgs() {
+    return {
+      avatarTemplate:
+        this.avatarTemplate || this.site.system_user_avatar_template,
```

Comment:
May look weird but basically if no user is associated we fallback to system avatar.